### PR TITLE
Add triggering from mavros

### PR DESCRIPTION
**Pull request by helenol**
Other features
 - Rip out dependencies (msgs) on other point grey drivers
 - Make camera start up NOT when the first message is subscribed
 - Remove unused publishing
 - Add trigger_delay to dynamic config
 - Timestamping is done mid-exposure (so turn on exposure output from driver)

To turn on, set the param `force_mavros_triggering` to true.

### CODE REVIEW

File: spinnaker_camera_driver/include/spinnaker_camera_driver/SpinnakerCamera.h [hunk 1]
```diff
@@ -163,6 +163,9 @@ class SpinnakerCamera {
   */
   void grabImage(sensor_msgs::Image* image, const std::string& frame_id);
 
+  // Gets the exposure of the last image, in seconds (?)
```
Comment (ZacharyTaylor) [inline spinnaker_camera_driver/include/spinnaker_camera_driver/SpinnakerCamera.h]: wasn't it in microseconds?

File: spinnaker_camera_driver/src/nodelet.cpp [hunk 2]
```diff
@@ -315,84 +315,133 @@ class SpinnakerCameraNodelet : public nodelet::Nodelet {
...
+      // Start the camera info manager and attempt to load any configurations
+      std::stringstream cinfo_name;
+      cinfo_name << serial;
+      cinfo_.reset(new camera_info_manager::CameraInfoManager(
+          nh, cinfo_name.str(), camera_info_url));
+
+      // Publish topics using ImageTransport through camera_info_manager (gives
+      // cool things like compression)
+      it_.reset(new image_transport::ImageTransport(nh));
+      // image_transport::SubscriberStatusCallback cb =
```
Comment (ZacharyTaylor) [inline spinnaker_camera_driver/src/nodelet.cpp]: remove commented code?

File: spinnaker_camera_driver/src/nodelet.cpp [hunk 3]
```diff
@@ -315,84 +315,133 @@ class SpinnakerCameraNodelet : public nodelet::Nodelet {
...
+    // First subscribe to the messages so we don't miss any.'
+    ros::NodeHandle& nh = getMTNodeHandle();
+    sequence_time_map_.clear();
+    trigger_sequence_offset_ = 0;
+
+    const std::string mavros_trigger_service = "/mavros/cmd/trigger_control";
+    if (ros::service::exists(mavros_trigger_service, false)) {
+      mavros_msgs::CommandTriggerControl req;
+      req.request.trigger_enable = true;
+      // This is NOT integration time, this is actually the sequence reset.
```
Comment (ZacharyTaylor) [inline spinnaker_camera_driver/src/nodelet.cpp:429]: great API

**Comment by ZacharyTaylor** [comment]
LGTM

File: spinnaker_camera_driver/include/spinnaker_camera_driver/SpinnakerCamera.h (hunk 1 above)
Comment (helenol) [reply spinnaker_camera_driver/include/spinnaker_camera_driver/SpinnakerCamera.h]: Yes great catch.

**Pull request merged**